services/common/src/components/projectSummary/AuthorizationsInvolved.tsx

```diff
@@ -48,45 +48,103 @@ import {
+  PROJECT_SUMMARY_DOCUMENT_TYPE_CODE,
+} from "../..";
 
 const RenderEMAPermitCommonSections = ({ code, isAmendment, index }) => {
   const dispatch = useDispatch();
   const purposeLabel = isAmendment
     ? "Additional Amendment Request Information"
     : "Purpose of Application";
-
+  const authType = isAmendment ? "AMENDMENT" : "NEW";
   const { authorizations, mine_guid, project_guid, project_summary_guid } = useSelector(
     getFormValues(FORM.ADD_EDIT_PROJECT_SUMMARY)
   );
-
   const codeAuthorizations = authorizations[code];
   const { AMENDMENT, NEW } = codeAuthorizations;
   const sectionValues = isAmendment ? AMENDMENT[index] : NEW[index];
-  const dfaRequired =
-    isAmendment &&
-    sectionValues?.amendment_changes?.some((val) => ["ILT", "IGT", "DDL"].includes(val));
-  const [showDocSection, setShowDocSection] = useState(sectionValues?.exemption_requested || false);
+  const [showExemptionSection, setshowExemptionSection] = useState(
+    sectionValues?.exemption_requested || false
+  );
 
   const projectSummaryDocumentTypesHash = useSelector(getProjectSummaryDocumentTypesHash);
 
   const onChange = (value, _newVal, _prevVal, _fieldName) => {
-    setShowDocSection(value);
+    setshowExemptionSection(value);
+  };
+
+  const isDocumentTypeRequired = (type) => {
+    let valuesToCheckFor = [];
+    if (type === "DFA") {
+      valuesToCheckFor = ["ILT", "IGT", "DDL"];
+    } else if (type === "CSL") {
+      valuesToCheckFor = ["TRA"];
+    } else if (type === "CON") {
+      valuesToCheckFor = ["TRA", "NAM"];
+    } else if (type === "CAF") {
+      valuesToCheckFor = ["MMR", "RCH"];
+    }
+
+    return sectionValues?.amendment_changes?.some((val) => valuesToCheckFor.includes(val));
+  };
+
+  const removeAmendmentDocument = (
+    amendmentDocumentsIndex: number,
+    category: string,
+    document_manager_guid: string
+  ) => {
+    const categorySpecificDocumentType = PROJECT_SUMMARY_DOCUMENT_TYPE_CODE_STATE[category];
+    const categorySpecificDocuments = sectionValues[categorySpecificDocumentType];
+    const categorySpecificDocumentIndex = categorySpecificDocuments.findIndex(
+      (doc) => document_manager_guid === doc.document_manager_guid
+    );
+
+    dispatch(
+      arrayRemove(
+        FORM.ADD_EDIT_PROJECT_SUMMARY,
+        `authorizations.[${code}].${authType}[${index}].${categorySpecificDocumentType}`,
+        categorySpecificDocumentIndex
+      )
+    );
+
+    dispatch(
+      arrayRemove(
+        FORM.ADD_EDIT_PROJECT_SUMMARY,
+        `authorizations.[${code}].${authType}[${index}].amendment_documents`,
+        amendmentDocumentsIndex
+      )
+    );
   };
 
-  const updateAmendmentDocuments = (documents: IProjectSummaryDocument[]) => {
-    const type = isAmendment ? "AMENDMENT" : "NEW";
+  const updateAmendmentDocuments = (document: IProjectSummaryDocument) => {
+    const category = document.category || document.project_summary_document_type_code;
+
+    dispatch(
+      arrayPush(
```

Comment:
I'm working with adding documents right now too, and I'm finding that if I do an arrayPush to the document field, that it doesn't register as touched (so validation doesn't change). If you're not running into that then don't worry about it, but just something to look out for.

Also I gotta get picky about this function name. It's no problem to change it from handling multiple documents to a single document BUT it should be renamed to reflect its purpose --> `updateAmendmentDocument`.

Reply:
The updateAmendmentDocument name has been changed now